### PR TITLE
Add tests for Create Sessions implementation

### PR DESCRIPTION
**Pull request by jonathanmendez**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Adds tests for the non-instrument-specific implementation of Create Sessions ( [AB#2520982](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2520982) )
- Adds a Fake IInstrument implementation that can be used to inject behavior and track calls (i.e. as a mock)

[CreateSessionsTestsPR.zip](https://github.com/ni/measurementlink-labview/files/12835066/CreateSessionsTestsPR.zip)

### Why should this Pull Request be merged?

We should test our code

### What testing has been done?

Ran all of these tests, including injection of failures to verify they are validly testing something.

### CODE REVIEW

File: Source/Tests/Tests.Runtime/Measurement Server/Helpers/Assertion Helpers/Assert Number of Calls.vi
Comment (pbirkhol-ni) [inline Source/Tests/Tests.Runtime/Measurement Server/Helpers/Assertion Helpers/Assert Number of Calls.vi:1]: Why does this add a newline to the result in the error case? I would expect the caller to add a newline, if necessary.

**Review by pbirkhol-ni — APPROVED** [review]

File: Source/Tests/Tests.Runtime/Measurement Server/Helpers/Fake Instrument/Fake Instrument.lvclass
[the remark this answers is not in the document]
Comment (jonathanmendez) [reply Source/Tests/Tests.Runtime/Measurement Server/Helpers/Fake Instrument/Fake Instrument.lvclass:1]: Done

File: Source/Tests/Tests.Runtime/Measurement Server/Create Sessions/test_EmptyInstrumentTypeId_CallCreateSessions_ReturnsError.vi
[the remark this answers is not in the document]
Comment (jonathanmendez) [reply Source/Tests/Tests.Runtime/Measurement Server/Create Sessions/test_EmptyInstrumentTypeId_CallCreateSessions_ReturnsError.vi:1]: Added assertion for empty array.

File: Source/Tests/Tests.Runtime/Measurement Server/Create Sessions/test_InstrumentTypeIdNotInReservation_CallCreateSessions_ReturnsErrorFor0Sessions.vi
[the remark this answers is not in the document]
Comment (jonathanmendez) [reply Source/Tests/Tests.Runtime/Measurement Server/Create Sessions/test_InstrumentTypeIdNotInReservation_CallCreateSessions_ReturnsErrorFor0Sessions.vi:1]: Good point. Done.

[the remark this answers is not in the document]
Comment (jonathanmendez) [reply Source/Tests/Tests.Runtime/Measurement Server/Create Sessions/test_InstrumentTypeIdNotInReservation_CallCreateSessions_ReturnsErrorFor0Sessions.vi:1]: Thanks, this was incorrect throughout. Updated the various tests.

File: Source/Tests/Tests.Runtime/Measurement Server/Helpers/Assertion Helpers/Assert Number of Calls.vi
Comment (jonathanmendez) [reply Source/Tests/Tests.Runtime/Measurement Server/Helpers/Assertion Helpers/Assert Number of Calls.vi:1]: Good catch. That was a result of a refactoring, but I think I'm just going to skip the newlines fully.

[the remark this answers is not in the document]
Comment (jonathanmendez) [reply Source/Tests/Tests.Runtime/Measurement Server/Helpers/Assertion Helpers/Assert Number of Calls.vi:1]: Done

**Pull request merged**